examples/idle-inhibit.c

```diff
@@ -0,0 +1,201 @@
+#include <GLES2/gl2.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <wayland-client.h>
+#include <wayland-egl.h>
+#include <wlr/render/egl.h>
+#include "xdg-shell-client-protocol.h"
+#include "idle-inhibit-unstable-v1-client-protocol.h"
+
+#include <linux/input-event-codes.h>
+
+/**
+ * Usage: idle-inhibit
+ * Creates a xdg-toplevel using the idle-inhibit protocol.
+ * It will be solid green, when it has an idle inhibitor, and solid yellow if
+ * it does not.
+ * Left click with a pointer will toggle this state. (Touch is not supported
+ * for now).
+ */
+
+static int width = 500, height = 300;
+
+static struct wl_compositor *compositor = NULL;
+static struct wl_seat *seat = NULL;
+static struct xdg_wm_base *wm_base = NULL;
+static struct zwp_idle_inhibit_manager_v1 *idle_inhibit_manager = NULL;
+static struct zwp_idle_inhibitor_v1 *idle_inhibitor = NULL;
+
+struct wlr_egl egl;
+struct wl_egl_window *egl_window;
+struct wlr_egl_surface *egl_surface;
+
+static void draw(void);
+
+static void pointer_handle_button(void *data, struct wl_pointer *pointer, uint32_t serial,
+		uint32_t time, uint32_t button, uint32_t state_w) {
+	struct wl_surface *surface = data;
+
+	if (button == BTN_LEFT && state_w == WL_POINTER_BUTTON_STATE_PRESSED) {
+		if (idle_inhibitor) {
+			zwp_idle_inhibitor_v1_destroy(idle_inhibitor);
+			idle_inhibitor = NULL;
+		} else {
+			idle_inhibitor =
+				zwp_idle_inhibit_manager_v1_create_inhibitor(
+					idle_inhibit_manager,
+					surface);
+		}
+	}
+
+	draw();
+}
+
+/* Function that just does nothing.
+ * When it is noop(void) (like draw) the compiler complains about type
+ * mismatches in the listener struct.
+ * Without any arguments, it can be implicitly casted
+ */
```

Comment:
I don't think this comment is necessary. I assume people should just know about that.

Reply:
Still, make it conform to the style guidelines

```c
static void noop() {
    /* This space intentionally left blank */
}
```

Also, don't these functions take arguments? You need to specify them or you might FUBAR the stack (depending on the target's ABI).

Reply:
I would asume the compiler implements it correctly, when it does not specify arguments, but I would not want to be quoted on that.
Let me check if I can find something, declaring multiple `noop` functions just feels silly to me.

I can't find anything in the wlroots nor the kernel coding style guids about the comment for empty blocks. Could we have that added for future reference?